WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift

```diff
@@ -875,121 +911,140 @@ import WordPressComAnalytics
 
 
     func syncItems(success:((hasMore: Bool) -> Void)?, failure: ((error: NSError) -> Void)?) {
+        guard let topic = readerTopic else {
+            DDLogSwift.logError("Error: Reader tried to sync items when the topic was nil.")
+            return
+        }
+
         let syncContext = ContextManager.sharedInstance().newDerivedContext()
         let service =  ReaderPostService(managedObjectContext: syncContext)
 
-        syncContext.performBlock {[weak self] () -> Void in
-            do {
-                let topic = try syncContext.existingObjectWithID(self!.readerTopic!.objectID) as! ReaderAbstractTopic
-                let objectID = topic.objectID
-                service.fetchPostsForTopic(topic,
-                    earlierThan: NSDate(),
-                    success: {[weak self] (count:Int, hasMore:Bool) in
-                        dispatch_async(dispatch_get_main_queue(), {
-                            if let strongSelf = self {
-                                if strongSelf.recentlyBlockedSitePostObjectIDs.count > 0 {
-                                    strongSelf.recentlyBlockedSitePostObjectIDs.removeAllObjects()
-                                    strongSelf.updateAndPerformFetchRequest()
-                                }
-                                strongSelf.updateLastSyncedForTopic(objectID)
+        syncContext.performBlock { [weak self] in
+            guard let topicInContext = (try? syncContext.existingObjectWithID(topic.objectID)) as? ReaderAbstractTopic else {
+                DDLogSwift.logError("Error: Could not retrieve an existing topic via its objectID")
+                return
+            }
+            let objectID = topicInContext.objectID
+            service.fetchPostsForTopic(topicInContext,
+                earlierThan: NSDate(),
+                success: { [weak self] (count:Int, hasMore:Bool) in
```

Comment:
Again, you could lose the type info and `count`, as it's not used:

`[weak self] _, hasMore in`